src/AssemblyLoader.cs

```diff
@@ -29,12 +31,14 @@ public class AssemblyLoader : IAssemblyLoader
         private const string GaugeLibAssembleName = "Gauge.CSharp.Lib";
         private readonly IAssemblyWrapper _assemblyWrapper;
         private readonly IReflectionWrapper _reflectionWrapper;
+        private readonly IStepRegistry _stepRegistry;
         private Assembly _targetLibAssembly;
 
-        public AssemblyLoader(IAssemblyWrapper assemblyWrapper, IEnumerable<string> assemblyLocations, IReflectionWrapper reflectionWrapper)
+        public AssemblyLoader(IStepRegistry registrty, IAssemblyWrapper assemblyWrapper, IEnumerable<string> assemblyLocations, IReflectionWrapper reflectionWrapper)
```

Comment:
`IStepRegistry` is injected as a dependency, but it is not one. Instead `AssemblyLoader` populates the `StepRegistry` in `UpdateStepRegistry`.

Instead, can we have `AssemblyLoader.GetStepRegistry` that instantiates and returns the `IStepRegistry`?

Mutating a dependency is a bad idea.

Reply:
done